File: scripts/GFS_IndicatorsTool/read_csv.py
import sys, json;
import time;
import pandas as pd;
import csv

inputFile = open(sys.argv[1] + '/input.json')
data = json.load(inputFile)
df = data['csv']
df= eval(df)

df = pd.DataFrame(df[1:], columns=df[0])

path= open(sys.argv[1] + '/obs.csv', "w")
df.to_csv(path)

dictionary = {
"csv_out": sys.argv[1] + '/obs.csv'
}


json_object = json.dumps(dictionary, indent = 2)
with open(sys.argv[1] + '/output.json', "w") as outfile:
    outfile.write(json_object)

File: scripts/GFS_IndicatorsTool/read_csv.R
library("rjson")


input <- fromJSON(file=file.path(outputFolder, "input.json"))

print(input)

tsv_json <-input$csv

##### Parse json text

### remove square brackets
tsv_json = substr(tsv_json, 3, nchar(tsv_json)-2)

### split lines
tsv_lines = strsplit(tsv_json, '\\), \\(')[[1]]

### convet to table
tsv_tab = do.call(rbind, strsplit(tsv_lines, ', '))

### convert to data.frame, set header
tsv_df = data.frame(tsv_tab[-1,])
colnames(tsv_df) = tsv_tab[1,]

colnames(tsv_df)[colnames(tsv_df)=="'decimal_latitude'"] = 'decimal_latitude'
colnames(tsv_df)[colnames(tsv_df)=="'decimal_longitude'"] = 'decimal_longitude'

### set coordinates as numeric
tsv_df$decimal_latitude = as.numeric(tsv_df$decimal_latitude)
tsv_df$decimal_longitude = as.numeric(tsv_df$decimal_longitude)

### set output path
tsv_out<-file.path(outputFolder, "species_obs.tsv")


### save output
write.table(tsv_df, file=tsv_out, sep='\t', quote=F, row.names = F, col.names = T)




### return output
output <- list("csv_out"=tsv_out)
print(output)

jsonData <- toJSON(output, indent=2)
write(jsonData, file.path(outputFolder,"output.json"))